src/cs50/flask.py

```diff
@@ -20,19 +22,42 @@
     except:
         pass
 
+    # Enable logging when Flask is in use,
+    # monkey-patching own SQL module, which shouldn't need to know about Flask
+    logging.getLogger("cs50").disabled = True
+    try:
+        import flask
+        from .sql import SQL
+        _before = SQL.execute
+        def _after(*args, **kwargs):
+            disabled = logging.getLogger("cs50").disabled
+            if flask.current_app:
+                logging.getLogger("cs50").disabled = False
+            try:
+                ret = _before(*args, **kwargs)
+                logging.getLogger("cs50").disabled = disabled
+                return ret
+            except:
+                logging.getLogger("cs50").disabled = disabled
+                raise
```

Comment:
I agree, empty `except`s are bad practice since e.g. they catch Ctrl-Cs (KeyboardInterrupt) etc. At a bare minimum you want `except Exception` which catches only exceptions that are meant to be caught, but really you should be specifying the exact kinds of exceptions you expect to be thrown.

The raising isn't meaningless though because this is inside the function definition for after

Reply:
Yeah, bad habit. But I blame Python's lack of documentation as to what exceptions can be thrown. :p